### PR TITLE
Add verify-index command

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change provides the necessary logic for parsing pgEdge cluster configurations as well as the command for reviewing PostgreSQL indexes across multiple databases to identify missing indexes. This also provides small additions to the README and Makefile.

Fixes https://mattermost.atlassian.net/browse/CLD-8419 and https://mattermost.atlassian.net/browse/CLD-8324

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+func New(dsn string, logger logrus.FieldLogger) (*SQLStore, error) {
+	url, err := url.Parse(dsn)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse dsn as an url")
+	}
+
+	var db *sqlx.DB
+
+	switch strings.ToLower(url.Scheme) {
+	case "postgres", "postgresql":
```
Comment (nickmisasi) [inline internal/store/store.go:33]: 👍

File: internal/store/store.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	DriverName() string
+}
+
+// exec executes the given query using positional arguments, automatically rebinding for the db.
+func (sqlStore *SQLStore) exec(e execer, sql string, args ...interface{}) (sql.Result, error) {
+	sql = sqlStore.db.Rebind(sql)
+	return e.Exec(sql, args...)
+}
+
+// exec executes the given query, building the necessary sql.
```
Comment (nickmisasi) [inline internal/store/store.go]:
```suggestion
// execBuilder executes the given query, building the necessary sql.
```

File: internal/store/store.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+func (sqlStore *SQLStore) execBuilder(e execer, b builder) (sql.Result, error) {
+	sql, args, err := b.ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build sql")
+	}
+
+	return sqlStore.exec(e, sql, args...)
+}
+
+// exec executes the given query, building the necessary sql.
```
Comment (nickmisasi) [inline internal/store/store.go]:
```suggestion
// Close closes the connection with the sqlStore
```

File: model/pgedge_config.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+type PgedgeNode struct {
+	Name      string `json:"name"`
+	IsActive  bool   `json:"is_active"`
+	IPAddress string `json:"ip_address"`
+	Port      int    `json:"port"`
+	Path      string `json:"path"`
+}
+
+func (dbc *PgedgeDatabaseConfig) Redact() {
```
Comment (nickmisasi) [inline model/pgedge_config.go]: Nit: would `Sanitize` be a more commonly used word for what this does?

File: README.md [hunk 5]
```diff
@@ -4,8 +4,18 @@ This repository contains tooling for reviewing Mattermost databases with complex
 
 ### Building
 
-To build and install, run the following:
+Before building the `mmdbt` CLI, ensure that you have the latest version of Go 1.22 installed.
```
Comment (nickmisasi) [inline README.md]: Should we link to the version in go.mod instead of explicitly stating 1.22, so that it's not missed in future version changes?

**Review by nickmisasi — APPROVED** [review]
A couple nits and suggestions but nothing that would block a merge. LGTM.

File: README.md (hunk 5 above)
Comment (gabrieljackson) [reply README.md]: Oh yeah, I like that idea. Let me change it.

File: README.md [hunk 6]
```diff
@@ -4,8 +4,18 @@ This repository contains tooling for reviewing Mattermost databases with complex
 
 ### Building
 
-To build and install, run the following:
+Before building the `mmdbt` CLI, ensure that you have Go installed and are using a version that it at least a new as the one [referenced here](https://github.com/mattermost/mattermost-dbt/blob/main/go.mod#L3).
```
Comment (nickmisasi) [inline README.md]:
```suggestion
Before building the `mmdbt` CLI, ensure that you have Go installed and are using a version that is at least as new as the one [referenced here](https://github.com/mattermost/mattermost-dbt/blob/main/go.mod#L3).
```

**Review by nickmisasi — APPROVED** [review]
One more nit. LGTM!

**Pull request merged**